vignettes/including-static-dashboards.Rmd

```diff
@@ -0,0 +1,126 @@
+vignette: >
+  %\VignetteIndexEntry{Static Dashboards}
+  %\VignetteEngine{knitr::rmarkdown}
+  %\VignetteEncoding{UTF-8}
+---
+
+Crunch allows you to host Shiny apps on the Crunch Jupyter server. Shiny apps are great, but they can be difficult to program, and can be very slow if not designed properly. [Flexdashboards](https://rmarkdown.rstudio.com/flexdashboard/) provide an easy-to-use alternative to Shiny apps. Instead of re-rendering dynamically based on the user action, Flexdashboards are effectively static websites that only include data which is live at the time of rendering. This can be desirable both because they are much easier to write, and because you can test and validate the dashboard's behavior before it goes to the client. Additionally, since you have more control over when the dashboard is rendered, you can include more computationally expensive operations like fitting a machine learning model without affecting user experience.
+
+You can host Flexdashboards on Crunch's Shiny infrastructure using Shiny's `includeHTML` function. What `includeHTML` does is add an HTML file to the shiny UI. If all you include in the shiny UI is the Flexdashboard html file, then the Shiny app effectively becomes a static site. 
+
+### Basic workflow
+Follow these steps to include a Flexdashboard in your Shiny app: 
+
+1) First create a new Rmarkdown document with a Flexdashboard output option
+1) Render the Rmd file into an html file
+1) Upload the rendered html file to your app's Jupyter folder
+1) In your shiny UI code add `includeHTML(<filename>)` to the `ui` object. Your `app.R` file might look something like this:
+
+```
+library(crunchy)
+ui <- fluidPage(
+    includeHTML("dashboard"))
+
+server <- function(input, output, session) {}
+
+# Run the application 
+shinyApp(ui = ui, server = server)
+```
+Note that we haven't included any authentication in this app which means that users who do not have access to a dataset could theoretically access it. The next step is to check that the user is logged into Crunch. 
+
+### Check User Authentication
+
+Crunchy includes a `shinyUser()` function which lets us display different content to the user depending on whether they are logged into Crunch of not. To do this we'll make use of Shiny's `renderUI` function to conditionally create a different UI for different types of users.
+
+```
+ui <- fluidPage(
+    uiOutput("dashboard")
+)
+```
+
+This creates the UI object, but most of the work is going to be done on the server side. We can use the `shinyUser()` function to get a user record for the current viewer of the app. This record includes some useful pieces of information, including the user's name, their email address, and their Crunch preferences. You can use this object to conditionally display different dashboards for different users like so: 
+
+```
+server <- function(input, output, session) {
+    user <- shinyUser()
+    output$dashboard <- renderUI({
+        if (is.character(email(user()))) {
```

Comment:
Use `checkAuthentication()` here. (Right now it's the same as `shinyUser()` but I don't plan on keeping it that way. We can add better messaging for when you're not authenticated, and try the pop-up or whatever to help you log in.)

The code we want here is 

```
output$dashboard <- renderUI({
    checkAuthentication()
    includeHTML("acme_dashboard.html")
})
```

If that doesn't work now because of reactivity (as in it would require `checkAuthentication()()`), let's make it work like that. We want to make it trivial to add an auth check.